source/val/validate_memory.cpp

```diff
@@ -405,6 +405,17 @@ spv_result_t ValidateVariable(ValidationState_t& _, const Instruction* inst) {
                << "must be typed as OpTypeStruct, or an array of this type";
       }
     }
+
+    if (storage_class == SpvStorageClassPushConstant) {
```

Comment:
Same here for the environment.

Reply:
Isn't this already gated above?
https://github.com/KhronosGroup/SPIRV-Tools/blob/70fdd2e3a0bc71918a1d34a31584077279557890/source/val/validate_memory.cpp#L377

I added a comment and reorganized according to the spec, if that helps.